### PR TITLE
add persistentVolume for static volumes

### PR DESCRIPTION
**Pull request by gertzakis**
### Fixes: #353 

With the proposed solution, it adds a `PersistentVolumeClaim` in order to be used as `volumeMount` in the `/opt/nautobot/static` directory of the Pods for storing the static files of Nautobot. 
This will give an easy solution of having a sigle init pod and the static files stored in a volume used by all the application pods

### CODE REVIEW

File: charts/nautobot/values.yaml [hunk 1]
```diff
@@ -236,6 +235,23 @@ nautobot:
             value: 2
             periodSeconds: 15
 
+  # Add a Persistent Volume Claim for /static files of Nautobot
+  persistence:
```
Comment (ubajze) [inline charts/nautobot/values.yaml:239]: I think the name here should be more explicit, as we only create a persistent volume for static files. For example, `persistenceStaticFiles`.

File: charts/nautobot/values.yaml [hunk 2]
```diff
@@ -236,6 +235,23 @@ nautobot:
...
+    # --- Enable PVC and the relevant volumes
+    enabled: false
+    # --- [Kubernetes StorageClass Name](https://kubernetes.io/docs/concepts/storage/storage-classes/), for the creation of Persistent Volume Claim
+    storageClass: ""
+    # --- [Kubernetes Volumes AccessMode](https://kubernetes.io/docs/concepts/storage/storage-classes/)
+    accessMode: "ReadWriteOnce"
+    # --- Persistent storage size request
+    size: "2Gi"
```
Comment (ubajze) [inline charts/nautobot/values.yaml:247]: Perhaps we can make this volume even smaller by default.

File: charts/nautobot/values.yaml [hunk 3]
```diff
@@ -236,6 +235,23 @@ nautobot:
             value: 2
             periodSeconds: 15
 
+  # Add a Persistent Volume Claim for /static files of Nautobot
```
Comment (ubajze) [inline charts/nautobot/values.yaml:238]:
```suggestion
  # Add a Persistent Volume Claim for Nautobot static files
```

File: charts/nautobot/templates/pvc-static.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{{- if .Values.nautobot.enabled -}}
+{{- if and .Values.nautobot.persistence.enabled -}}
```
Comment (ubajze) [inline charts/nautobot/templates/pvc-static.yaml:2]: Do we need `and` here?

File: charts/nautobot/values.yaml [hunk 5]
```diff
@@ -236,6 +235,23 @@ nautobot:
...
+  persistence:
+    # --- Enable PVC and the relevant volumes
+    enabled: false
+    # --- [Kubernetes StorageClass Name](https://kubernetes.io/docs/concepts/storage/storage-classes/), for the creation of Persistent Volume Claim
+    storageClass: ""
```
Comment (ubajze) [inline charts/nautobot/values.yaml:243]: Should we change this to `storageClassName` to be more explicit?

File: docs/advanced-features/persistence.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# Persistence for Static files
+
+Besides using an external shared storage (e.g. S3) for storing static files, it is also possible to leverage Kubernetes persistence. The following configuration creates a Persistent Volume Claim called `nautobot-static` and mounts it at the `/opt/nautobot/static` path of the Pods.
```
Comment (ubajze) [inline docs/advanced-features/persistence.md:3]:
```suggestion
Besides using an external shared storage (e.g. S3) for storing static files, it is also possible to leverage Kubernetes persistence. The following configuration creates a Persistent Volume Claim and mounts it at the `/opt/nautobot/static` path of the Pods.
```

It is not necessary that PVC name is `nautobot-static`

File: docs/advanced-features/persistence.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
+# Persistence for Static files
```
Comment (ubajze) [inline docs/advanced-features/persistence.md:1]: Can we add somewhere that using storage such as S3 is still a recommended way when running a deployment across multiple Kubernetes nodes?

File: mkdocs.yml [hunk 8]
```diff
@@ -83,6 +83,7 @@ nav:
     - Ingress: "advanced-features/ingress.md"
     - MySQL Support: "advanced-features/mysql.md"
     - Nautobot as a Subchart: "advanced-features/nautobot-as-subchart.md"
+    - Persistent Volume: "advanced-features/persistence.md"
```
Comment (ubajze) [inline mkdocs.yml:86]:
```suggestion
    - Persistent Volume for Static Files: "advanced-features/persistence.md"
```

File: charts/nautobot/values.yaml [hunk 9]
```diff
@@ -236,6 +235,23 @@ nautobot:
...
+    # --- [Kubernetes Volumes AccessMode](https://kubernetes.io/docs/concepts/storage/storage-classes/)
+    accessMode: "ReadWriteOnce"
+    # --- Persistent storage size request
+    size: "2Gi"
+    # ---
```
Comment (ubajze) [inline charts/nautobot/values.yaml:248]: Can we add a text here as well?

File: docs/advanced-features/index.md [hunk 10]
```diff
@@ -14,6 +14,7 @@ With [Kubernetes](https://kubernetes.io/), public clouds, and the [dependency su
 * [Ingress](ingress/)
 * [MySQL Support](mysql/)
 * [Nautobot as a Subchart](nautobot-as-subchart/)
+* [Persistent Volume](persistence/)
```
Comment (ubajze) [inline docs/advanced-features/index.md:17]:
```suggestion
* [Persistent Volume for Static Files](persistence/)
```

File: charts/nautobot/values.yaml (hunk 1 above)
Comment (gertzakis) [reply charts/nautobot/values.yaml:239]: Agreed! Changed it

File: charts/nautobot/values.yaml (hunk 2 above)
Comment (gertzakis) [reply charts/nautobot/values.yaml:247]: Made it `1Gi`

File: charts/nautobot/values.yaml (hunk 3 above)
Comment (gertzakis) [reply charts/nautobot/values.yaml:238]: Changed it

File: charts/nautobot/templates/pvc-static.yaml (hunk 4 above)
Comment (gertzakis) [reply charts/nautobot/templates/pvc-static.yaml:2]: Nice catch

File: charts/nautobot/values.yaml (hunk 5 above)
Comment (gertzakis) [reply charts/nautobot/values.yaml:243]: Agreed, changed it to `storageClassName`

File: docs/advanced-features/persistence.md (hunk 6 above)
Comment (gertzakis) [reply docs/advanced-features/persistence.md:3]: Changed it

File: mkdocs.yml (hunk 8 above)
Comment (gertzakis) [reply mkdocs.yml:86]: Changed it, also in `index.md`

File: docs/advanced-features/index.md (hunk 10 above)
Comment (gertzakis) [reply docs/advanced-features/index.md:17]: Changed it

File: charts/nautobot/values.yaml (hunk 9 above)
Comment (gertzakis) [reply charts/nautobot/values.yaml:248]: Nice catch!

File: docs/advanced-features/persistence.md (hunk 7 above)
Comment (gertzakis) [reply docs/advanced-features/persistence.md:1]: Sure, changed the docs. Please check them again!

File: charts/nautobot/values.schema.json [hunk 11]
```diff
@@ -70,6 +70,35 @@
             }
           }
         },
+        "persistence": {
```
Comment (ubajze) [inline charts/nautobot/values.schema.json:73]:
```suggestion
        "persistenceStaticFiles": {
```

File: charts/nautobot/values.schema.json [hunk 12]
```diff
@@ -70,6 +70,35 @@
...
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "enabled": {
+              "type": "boolean"
+            },
+            "storageClass": {
```
Comment (ubajze) [inline charts/nautobot/values.schema.json:80]:
```suggestion
            "storageClassName": {
```

File: charts/nautobot/values.schema.json [hunk 13]
```diff
@@ -70,6 +70,35 @@
...
+            "selector": {
+              "type": "object"
+            }
+          },
+          "required": [
+            "accessMode",
+            "enabled",
+            "selector",
+            "size",
+            "storageClass"
```
Comment (ubajze) [inline charts/nautobot/values.schema.json:98]:
```suggestion
            "storageClassName"
```

**Review by ubajze — APPROVED** [review]

**Review by ubajze — APPROVED** [review]

**Pull request merged**